### PR TITLE
ESP32: Add cmake support to temperature-measurement-app and persistent-storage.

### PR DESCRIPTION
**Pull request by sweetymhaiske**
#### Problem
 Add CMake support to esp32 temperature-measurement-app and persistent-storage.

 #### Summary of Changes
1. Added `CMakeLists.txt`.
2. Updated `README.md.`

### CODE REVIEW

**Review by msandstedt — APPROVED** [review]

File: examples/all-clusters-app/esp32/README.md [hunk 1]
```diff
@@ -77,13 +71,9 @@ make sure the IDF_PATH has been exported(See the manual setup steps above).
...
 
-        To use the default configuration options, run the default config
-
-          $ idf make defconfig
+-   To build the demo application.
 
--   Run make to build the demo application.
-
-          $ idf make
+          $ idf.py build
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/esp32/README.md:76]: Does that automatically select reasonable `-j` options or equivalent?

That is, how do I do the equivalent of `idf make -j16 flash monitor ESPPORT=/dev/tty.SLAB_USBtoUART` in this new setup?

File: examples/all-clusters-app/esp32/README.md [hunk 2]
```diff
@@ -53,17 +53,11 @@ step. To install these components manually, follow these steps:
...
-    builds.
-
-          $ source idf.sh
-
 -   Configuration Options
 
         To choose from the different configuration options, run menuconfig
 
-          $ idf make menuconfig
+          $ idf.py menuconfig
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/esp32/README.md:60]: Something above should talk about setting up PATH so this is in it, right?

File: examples/persistent-storage/esp32/README.md [hunk 3]
```diff
@@ -59,21 +59,15 @@ step. To install these components manually, follow these steps:
...
-
-          $ source idf.sh
-
 -   Configuration Options
 
-        To use the default configuration options, run the default config
+        To choose from the different configuration options, run menuconfig
 
-          $ idf make defconfig
+          $ idf.py menuconfig
```
Comment (bzbarsky-apple) [inline examples/persistent-storage/esp32/README.md:66]: Similar comments here as for the all-clusters-app README.

File: examples/temperature-measurement-app/esp32/README.md [hunk 4]
```diff
@@ -28,17 +28,11 @@ step. To install these components manually, follow these steps:
...
-    builds.
-
-          $ source idf.sh
-
 -   Configuration Options
 
         To choose from the different configuration options, run menuconfig
 
-          $ idf make menuconfig
+          $ idf.py menuconfig
```
Comment (bzbarsky-apple) [inline examples/temperature-measurement-app/esp32/README.md:35]: And here, same comments.

File: examples/all-clusters-app/esp32/README.md (hunk 1 above)
Comment (dhrishi) [reply examples/all-clusters-app/esp32/README.md:76]: @bzbarsky-apple I think, idf.py should internally take care of parallel builds. I will confirm this internally and get back.

File: examples/all-clusters-app/esp32/README.md (hunk 2 above)
Comment (dhrishi) [reply examples/all-clusters-app/esp32/README.md:60]: @sweetymhaiske  In all the examples, the `./install.sh` should be followed by `. ./export.sh` command

File: examples/persistent-storage/esp32/sdkconfig.defaults [hunk 5]
```diff
@@ -25,3 +25,6 @@ CONFIG_ESPTOOLPY_MONITOR_BAUD=115200
 # Use a custom partition table
 CONFIG_PARTITION_TABLE_CUSTOM=y
 CONFIG_PARTITION_TABLE_FILENAME="partitions.csv"
+
+#Enable BT
+CONFIG_BT_ENABLED=y
```
Comment (dhrishi) [inline examples/persistent-storage/esp32/sdkconfig.defaults:30]: Why is BT enabled in `persistent-storage` example?

**Review by dhrishi — CHANGES_REQUESTED** [review]

File: examples/all-clusters-app/esp32/README.md (hunk 1 above)
Comment (dhrishi) [reply examples/all-clusters-app/esp32/README.md:76]: @bzbarsky-apple Yes, confirmed, idf.py handles it internally. You need not add anything else

File: examples/all-clusters-app/esp32/README.md (hunk 2 above)
Comment (sweetymhaiske) [reply examples/all-clusters-app/esp32/README.md:69]: Added.

File: examples/persistent-storage/esp32/README.md (hunk 3 above)
Comment (sweetymhaiske) [reply examples/persistent-storage/esp32/README.md:75]: Added.

File: examples/temperature-measurement-app/esp32/README.md (hunk 4 above)
Comment (sweetymhaiske) [reply examples/temperature-measurement-app/esp32/README.md:44]: Added.

File: examples/persistent-storage/esp32/sdkconfig.defaults (hunk 5 above)
Comment (sweetymhaiske) [reply examples/persistent-storage/esp32/sdkconfig.defaults]: Fixed.

**Review by woody-apple — APPROVED** [review]

File: examples/all-clusters-app/esp32/README.md [hunk 6]
```diff
@@ -53,17 +53,30 @@ step. To install these components manually, follow these steps:
...
--   In the root of the example directory, sync the CHIP tree's submodules and
-    source `idf.sh`. Note: This does not have to be repeated for incremental
-    builds.
+-   Setting up the environment
 
-          $ source idf.sh
+To download and install packages run bootstrap.sh
+
+        $ cd ${HOME}/tools/esp-idf
+        $ . ./export.sh
```
Comment (dhrishi) [inline examples/all-clusters-app/esp32/README.md:61]: Do you also want to add `./install.sh` ?

Comment (dhrishi) [reply examples/all-clusters-app/esp32/README.md:61]: Please add this to other examples as well

File: examples/all-clusters-app/esp32/README.md [hunk 7]
```diff
@@ -53,17 +53,30 @@ step. To install these components manually, follow these steps:
 Currently building in VSCode _and_ deploying from native is not supported, so
 make sure the IDF_PATH has been exported(See the manual setup steps above).
 
--   In the root of the example directory, sync the CHIP tree's submodules and
-    source `idf.sh`. Note: This does not have to be repeated for incremental
-    builds.
+-   Setting up the environment
 
-          $ source idf.sh
+To download and install packages run bootstrap.sh
```
Comment (dhrishi) [inline examples/all-clusters-app/esp32/README.md:58]: Nitpick: Remove `run bootstrap.sh` from here? Applicable to all examples

**Review by dhrishi — CHANGES_REQUESTED** [review]

File: examples/all-clusters-app/esp32/README.md (hunk 6 above)
Comment (sweetymhaiske) [reply examples/all-clusters-app/esp32/README.md:61]: It is already mentioned while setting up `esp-idf`.

File: examples/all-clusters-app/esp32/README.md (hunk 7 above)
Comment (sweetymhaiske) [reply examples/all-clusters-app/esp32/README.md]: Done.

File: config/esp32/components/chip/CMakeLists.txt [hunk 8]
```diff
@@ -30,12 +30,7 @@ if(NOT CHIP_ROOT)
 endif()
 
 idf_component_register(SRCS chip.c chip.cpp
-                       PRIV_REQUIRES freertos lwip bt mdns mbedtls)
-
-# Override some build specifications
-set_property(TARGET ${COMPONENT_LIB} PROPERTY CXX_STANDARD 14)
-set(CMAKE_BUILD_TYPE MinSizeRel)
-target_compile_options(${COMPONENT_LIB} PRIVATE "-DLWIP_IPV6_SCOPES=0" "-DCHIP_HAVE_CONFIG_H")
```
Comment (dhrishi) [inline config/esp32/components/chip/CMakeLists.txt:38]: @sweetymhaiske Where are the corresponding new additions?

File: config/esp32/components/chip/CMakeLists.txt [hunk 9]
```diff
@@ -30,12 +30,7 @@ if(NOT CHIP_ROOT)
...
+                       PRIV_REQUIRES freertos fatfs lwip bt mdns mbedtls)
```
Comment (dhrishi) [inline config/esp32/components/chip/CMakeLists.txt:33]: @sweetymhaiske Why is fatfs required to be added?

File: config/esp32/components/chip/CMakeLists.txt (hunk 8 above)
Comment (sweetymhaiske) [reply config/esp32/components/chip/CMakeLists.txt:38]: Yes, that's how it is done for Makefile too.

File: config/esp32/components/chip/CMakeLists.txt (hunk 9 above)
Comment (sweetymhaiske) [reply config/esp32/components/chip/CMakeLists.txt:33]: It is needed for pigweed, but for now, CMake support is not added for pigweed hence I am removing it.

File: examples/all-clusters-app/esp32/README.md (hunk 6 above)
Comment (dhrishi) [reply examples/all-clusters-app/esp32/README.md:61]: Still it will be better if you add that. It is present in the current README.md

Comment (sweetymhaiske) [reply examples/all-clusters-app/esp32/README.md:62]: Added.

**Review by dhrishi — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
It looks like idf.py doesn't work for building all-clusters-app, at least for me... I filed https://github.com/project-chip/connectedhomeip/issues/5818

**Pull request merged**